buildingmotif/dataclasses/template.py

```diff
@@ -313,6 +340,10 @@ def evaluate(
                 for arg in unbound_optional_args:
                     remove_triples_with_node(templ.body, PARAM[arg])
             return templ.body
+        if len(leftover_params) > 0 and warn_unused:
+            warnings.warn(
```

Comment:
`logging.warning` doesn't actually raise a warning, which is the behavior I wanted because it's more explicit than a logging message:

```python
import pytest
import logging
import warning

with pytest.warns(Warning):
  warning.warn('hi') # works

with pytest.warns(Warning):
  logging.warning('hi') # fails
```

Open to other ideas though!

Reply:
oh cool. Well, since warning is a builtin package I don't really have strong feelings